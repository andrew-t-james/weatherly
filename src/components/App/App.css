.seven-hour {
  background-color: var( --color-blue-light-1);
}