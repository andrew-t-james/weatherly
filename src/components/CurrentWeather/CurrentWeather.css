.current-weather {
  align-items: center;
  border-bottom: var(--line);
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 2rem;
}

.current-weather__title {
  font-size: 3.5rem;
  font-weight: 400;
}
.current-weather__weather,
.current-weather__temperature,
.current-weather__description,
.current-weather__high,
.current-weather__low {
  color: var(--color-grey-dark-1);
  font-weight: 400;
}

.current-weather__weather,
.current-weather__temperature,
.current-weather__description {
  font-size: 2rem;
}

.current-weather__description {
  font-size: 1.7rem;
}

.current-weather__icon {
  font-size: 8rem;
  margin: 2rem 0;
}

.current-weather__high,
.current-weather__low {
  font-size: 1.8rem;
}

.current-weather__description {
  margin-top: 1rem;
}