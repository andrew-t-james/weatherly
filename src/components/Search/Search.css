.header {
  align-items: center;
  display: flex;
  justify-content: space-around;
}

/* .search-form {
  align-items: center;
  display: flex;
  flex: 0 0 40%;
  justify-content: center;
}

.search-form__input {
  background-color: var(--color-grey-light-2);
  border: none;
  border-radius: 100px;
  color: inherit;
  font-family: inherit;
  font-size: inherit;
  margin-right: -5.25rem;
  padding: 0.7rem 2rem;
  transition: all 0.2s;
  width: 90%;
}

.search-form__input:focus {
  background-color: var(--color-grey-light-3);
  outline: none;
  width: 100%;
}

.search-form__input::-webkit-input-placeholder {
  color: var(--color-grey-light-4);
  font-weight: 100;
}

.search-form__input:focus + .search-form__button {
  background-color: var(--color-grey-light-3);
}

.search-form__button {
  background-color: var(--color-grey-light-2);
  border: none;
  text-transform: uppercase;
}

.search-form__button:focus {
  outline: none;
}

.search-form__button:hover {
  cursor: pointer;
}

.search-form__button:active {
  transform: translateY(2px);
}

input:-webkit-autofill {
  -webkit-box-shadow: 0 0 0 30px white inset;
} */