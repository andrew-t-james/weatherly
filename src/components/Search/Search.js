import React, { Component } from 'react';

class Search extends Component {
  constructor() {
    super();
    this.state = {
      location: ''
    };

    this.updateInput = this.updateInput.bind(this);
    this.submitLocation = this.submitLocation.bind(this);
  }

  updateInput(event) {
    const { name, value } = event.target;

    this.setState({
      [name]: value
    });
  }

  submitLocation(event) {
    const { location } = this.state;

    event.preventDefault();
    this.props.updateLocation(location);

    this.setState({
      location: ''
    });

    event.target.reset();
  }

  render() {
    const  { location } = this.state;

    return (
      <header>
        <h1>Weathrly App</h1>
        <form onSubmit={event => this.submitLocation(event)}>
          <input
            onChange={event => this.updateInput(event)}
            name="location" type="text"
            value={location}/>
          <button type="submit" disabled={!location.trim(' ')}>Submit</button>
        </form>
      </header>
    );
  }
}

export default Search;