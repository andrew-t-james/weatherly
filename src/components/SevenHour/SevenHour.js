import React from 'react';
import ForecastCard from '../ForecastCard/ForecastCard.js';

const SevenHour = (props) => {
  const { hourly } = props;

  return (
    <div className="forecast">
      <h1 className="forecast-title">Seven Hour</h1>
      <ul className="forecast-list">
        {
          hourly.map((hour, index) =>
            <ForecastCard
              key={index}
              forecast={hour}/>
          )
        }
      </ul>
    </div>
  );
};

export default SevenHour;