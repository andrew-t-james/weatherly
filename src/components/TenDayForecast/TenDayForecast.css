.forecast {
  border-bottom: var(--line);
  padding: 2rem 0;
  width: 100%;
}

.forecast-list {
  display: flex;
  flex-direction: column;
  font-size: 1.4rem;
  justify-content: space-around;
  list-style: none;
  text-align: center;
}

@media screen and  (min-width: 560px) {
  .forecast-list {
    flex-direction: row;
    flex-wrap: wrap;
  }
}