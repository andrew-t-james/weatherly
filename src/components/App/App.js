import React, { Component } from 'react';
import CurrentWeather from '../CurrentWeather/CurrentWeather.js';
import SevenDayForecast from '../SevenDayForecast/SevenDayForecast.js';

import { data, dailyForecast } from '../../mock-data/mock-data.js';

class App extends Component {

  render() {
    return (
      <div>
        <h1>Weathrly App</h1>
        <CurrentWeather
          currentWeather={data.current_observation} />
        <SevenDayForecast
          dailyForecast={dailyForecast} />
      </div>
    );
  }
}

export default App;