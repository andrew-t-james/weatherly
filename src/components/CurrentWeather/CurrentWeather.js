import React from 'react';

import './CurrentWeather.css';

const CurrentWeather = (props) => {
  const { city, icon, iconUrl, temperature, weather, high, low, description } = props.currentWeather;

  return (
    <div className="current-weather">
      <h2 className="current-weather__title">{city}</h2>
      <p className="current-weather__weather">{weather}</p>
      <p className="current-weather__temperature">{temperature}</p>
      <img className="current-weather__icon" src={iconUrl} alt={icon}/>
      <div className="high-low">
        <p className="current-weather__high">{high} &uarr;</p>
        <p className="current-weather__low">{low} &darr;</p>
      </div>
      <p className="current-weather__description">{description}</p>
    </div>
  );
};

export default CurrentWeather;
