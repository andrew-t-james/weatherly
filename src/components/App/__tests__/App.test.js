import React from 'react';
import { shallow } from 'enzyme';
import App from '../App.js';
import Search from '../../Search/Search.js';
import TenDayForecast from '../../TenDayForecast/TenDayForecast.js';
import CurrentWeather from '../../CurrentWeather/CurrentWeather.js';
import SevenHour from '../../SevenHour/SevenHour.js';
import ErrorModal from '../../ErrorModal/ErrorModal.js';
import { API_KEY } from '../../../API_KEY.js';
import mockJSON from '../../../mock-data/mock-data.json';

describe('App unit tests', () => {
  let app;

  beforeEach(() => app = shallow(<App />));

  test('It should have a default values of null for currentLocations values', () => {
    const expected = {
      city: null,
      weather: null,
      icon: null,
      iconUrl: null,
      temperature: null,
      high: null,
      low: null,
      description: null
    };

    const actual = app.state('currentLocation');

    expect(actual).toEqual(expected);
  });

  test('It should have a default state of [] for sevenHourForecast', () => {
    const expected = [];

    const actual = app.state('sevenHourForecast');

    expect(actual).toEqual(expected);
  });

  test('It should have a default state of [] for tenDayForecast', () => {
    const expected = [];

    const actual = app.state('tenDayForecast');

    expect(actual).toEqual(expected);
  });

  test('App should render a single TenDayForecast component', () => {
    const expected = 1;
    const actual = app.find(Search).length;

    expect(actual).toBe(expected);
  });

  test('App should render a single CurrentWeather component', () => {
    const expected = 1;
    const actual = app.find(CurrentWeather).length;

    expect(actual).toBe(expected);
  });

  test('App should render a single SevenHour component', () => {
    const expected = 1;
    const actual = app.find(SevenHour).length;

    expect(actual).toBe(expected);
  });

  test('App should render a single TenDayForecast component', () => {
    const expected = 1;
    const actual = app.find(TenDayForecast).length;

    expect(actual).toBe(expected);
  });

  test('should display an errorModal if hasError is true', async () => {
    const expected = 1;

    app.setState({
      hasError: true
    });

    const actual = app.find(ErrorModal).length;

    expect(actual).toBe(expected);
  });

  describe('API Unit tests', () => {
    test('should have been called with correct url', async () => {
      const apiEndPoint = `http://api.wunderground.com/api/${API_KEY}/forecast/forecast10day/conditions/hourly/q/KY/Louisville.json`;

      window.fetch = jest.fn().mockImplementation(() =>
        Promise.resolve({
          status: 200,
          json: () => Promise.resolve(mockJSON)
        }));

      await app.instance().updateLocation('Louisville');

      expect(window.fetch).toHaveBeenCalledWith(apiEndPoint);
    });
  });
});