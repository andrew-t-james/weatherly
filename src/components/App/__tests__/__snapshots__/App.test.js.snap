// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App unit tests it should match snapshot 1`] = `
<main>
  <header>
    <h1>
      Weathrly App
    </h1>
    <form
      onSubmit={[Function]}
    >
      <input
        name="location"
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        disabled={true}
        type="submit"
      >
        Submit
      </button>
    </form>
  </header>
  <div
    className="current-weather"
  >
    <h2
      className="current-weather__title"
    />
    <p
      className="current-weather__weather"
    />
    <p
      className="current-weather__temperature"
    />
    <img
      alt={null}
      className="current-weather__icon"
      src={null}
    />
    <div
      className="high-low"
    />
  </div>
  <section
    className="forecast"
  >
    <h1
      className="forecast-title"
    >
      Seven Hour
    </h1>
    <ul
      className="forecast-list"
    />
  </section>
  <section
    className="forecast"
  >
    <h1
      className="forecast-title"
    >
      Ten Day Forecast
    </h1>
    <ul
      className="forecast-list"
    />
  </section>
</main>
`;
